AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -25,6 +26,41 @@ static void runLoopSourceCallback(void *info) {
 #endif
 }
 
+#pragma mark - ASDeallocThread
+
+@interface ASDeallocThread : NSThread
+@property (nonatomic, strong) ASRunLoopQueue *deallocQueue;
+@end
+
+@implementation ASDeallocThread
+
+- (void)main
+{
+  self.name = @"ASDeallocThread";
+  [self deallocQueue];
+  while (YES) {
+    // This method will still return once all sources have finished.
+    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate distantFuture]];
```

Comment:
@maicki my current implementation is using the runloop management in ASRunLoopQueue.  The NSThread exists so it has a single-threaded place to perform the work, without the deallocations going onto the main queue runloop.  I don't think I can get a runloop reference to provide to the ASRunLoopQueue unless I have the thread...